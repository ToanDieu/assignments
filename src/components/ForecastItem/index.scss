.forecast-item {
  width: 100%;
  &__title {
    color: #337ab7;
    font-weight: 500;
    display: block;
    width: 100%;
    text-align: initial;
  }
  &__status {
    display: flex;
    align-items: baseline;
    margin: 20px 0;
    .status {
      &__icon {
        width: 32px;
        height: 32px;
        margin: 0 5px 0 0;
      }
      &__text {
        font-size: 13px;
      }
    }
  }
  &__celsius {
    display: flex;
    font-size: 13px;
  }
}
