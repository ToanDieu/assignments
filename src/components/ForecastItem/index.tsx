import "./index.scss";

type Props = {
  applicable_date: string;
  max_temp: number;
  min_temp: number;
  weather_state_abbr: string;
  weather_state_name: string;
};

export default function ForecastItem(props: Props) {
  return (
    <div className={"forecast-item"}>
      <span className={"forecast-item__title"}>{props.applicable_date}</span>
      <div className={"forecast-item__status status"}>
        <img
          className={"status__icon"}
          alt={props.weather_state_abbr}
          src={`https://www.metaweather.com//static/img/weather/${props.weather_state_abbr}.svg`}
        />
        <span className={"status__text"}>{props.weather_state_name}</span>
      </div>
      <span className={"forecast-item__celsius"}>
        {`Max: ${props.max_temp} `}&#8451;
      </span>
      <span className={"forecast-item__celsius"}>
        {`Min: ${props.min_temp} `}&#8451;
      </span>
    </div>
  );
}
