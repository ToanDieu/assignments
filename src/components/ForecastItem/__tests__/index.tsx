import { render, cleanup } from "@testing-library/react";

import ForecastItem from "../index";

afterEach(cleanup);

describe("<ForecastItem/>", () => {
  const applicable_date = "today";

  const weather_state_abbr = "lr";
  const weather_state_name = "light rain";

  const max_temp = 10;
  const min_temp = 8;
  test("item should render correct passed values", () => {
    const { container } = render(
      <ForecastItem
        applicable_date={applicable_date}
        max_temp={max_temp}
        min_temp={min_temp}
        weather_state_abbr={weather_state_abbr}
        weather_state_name={weather_state_name}
      />
    );
    const titleSpan = container.querySelector("span.forecast-item__title");

    const statusDesSpan = container.querySelector("span.status__text");
    const imgStatus = container.querySelector("img.status__icon");

    const tempSpans = container.querySelectorAll("span.forecast-item__celsius");

    expect(titleSpan?.textContent).toBe(applicable_date);

    expect(statusDesSpan?.textContent).toBe(weather_state_name);
    expect(imgStatus).toHaveProperty(
      "src",
      `https://www.metaweather.com//static/img/weather/${weather_state_abbr}.svg`
    );

    expect(tempSpans[0].textContent).toContain(max_temp);
    expect(tempSpans[1].textContent).toContain(min_temp);
  });

  test("should render and match snapshort", () => {
    const { container } = render(
      <ForecastItem
        applicable_date={applicable_date}
        max_temp={max_temp}
        min_temp={min_temp}
        weather_state_abbr={weather_state_abbr}
        weather_state_name={weather_state_name}
      />
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
