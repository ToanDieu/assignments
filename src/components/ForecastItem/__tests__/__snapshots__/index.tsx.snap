// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ForecastItem/> should render and match snapshort 1`] = `
<div
  class="forecast-item"
>
  <span
    class="forecast-item__title"
  >
    today
  </span>
  <div
    class="forecast-item__status status"
  >
    <img
      alt="lr"
      class="status__icon"
      src="https://www.metaweather.com//static/img/weather/lr.svg"
    />
    <span
      class="status__text"
    >
      light rain
    </span>
  </div>
  <span
    class="forecast-item__celsius"
  >
    Max: 10 
    ℃
  </span>
  <span
    class="forecast-item__celsius"
  >
    Min: 8 
    ℃
  </span>
</div>
`;
